include/stdexec/functional.hpp

```diff
@@ -77,6 +77,12 @@ namespace stdexec {
           tag_invoke((_Tag&&) __tag, (_Args&&) __args...);
         };
 
+    template <class _Ret, class _Tag, class... _Args>
+      concept __tag_invocable_r =
+        requires (_Tag __tag, _Args&&... __args) {
+          { tag_invoke((_Tag&&) __tag, (_Args&&) __args...) } -> same_as<_Ret>;
```

Comment:
Let's make this behave like `std::is_invocable_r`. The constraint should be `convertible_to<_Ret>`, except that if `_Ret` is `void`, any return type will do.

Reply:
Yep

Reply:
I suppose you could just check the expression:

```c++
static_cast<_Ret>(tag_invoke((_Tag&&) __tag, (_Args&&) __args...));
```